### PR TITLE
export common env vars in bash scripts

### PR DESCRIPTION
**Pull request by phunehehe**

### CODE REVIEW

File: pkgs/default.nix [hunk 1]
```diff
@@ -104,16 +120,20 @@ let
...
+          # set -u # We are not ready for this yet
+          ${exportSessionVariables}
+        '';
       in
-      pkgs.runCommand name { inherit script; } ''
-        echo '#!${pkgs.bash}/bin/bash' > "$out"
-        echo 'set -e' >> "$out"
-        #echo 'set -u' >> "$out" # we are not ready for this yet
-        echo 'set -o pipefail' >> "$out"
-        echo -n "$script" >> "$out"
```
Comment (phunehehe) [inline pkgs/default.nix:113]: @4z3 I think `echo -n` is an annoyance (it messes up the output of `cat`). Is there any reason you want to keep it?

Comment (4z3) [inline pkgs/default.nix:113]: `echo -n` didn't originate from me and I don't care about it :)

File: pkgs/default.nix [hunk 2]
```diff
@@ -5,15 +5,29 @@
...
+    fetchurl newScope;
   inherit (lib)
-    concatMapStringsSep overrideDerivation optionalAttrs makeSearchPath;
+    concatMapStringsSep overrideDerivation optionalAttrs makeSearchPath
+    concatStringsSep;
 
   fns = rec {
 
+
+    exportSessionVariables = mapcats (k: v: "export ${k}=${v}") {
```
Comment (proger) [inline pkgs/default.nix]: this one looks pure

Comment (phunehehe) [inline pkgs/default.nix]: okay I'll admit I'm confused by `fns` (functions?) that aren't meant to be pure
but it makes sense to move this one to lib, now that it exists

Comment (phunehehe) [inline pkgs/default.nix]: moved

File: lib/default.nix [hunk 3]
```diff
@@ -0,0 +1,18 @@
+let
+  inherit (import <microgram/sdk.nix>) pkgs lib;
+  inherit (lib) mapAttrsToList concatStringsSep;
+in rec {
+
+  exportSessionVariables = mapcats (k: v: "export ${k}=${v}") {
+
+    TZDIR = "${pkgs.tzdata}/share/zoneinfo";
```
Comment (proger) [inline lib/default.nix]: can we drag tzdata through sdk?

Comment (phunehehe) [inline lib/default.nix]: It is via sdk. Or how else do you mean?

Comment (proger) [inline lib/default.nix]: `pkgs` here is a direct reference to nixpkgs, that's mostly sticking out for debugging purposes

Comment (phunehehe) [inline lib/default.nix]: I didn't realize that. Changed.

**Pull request merged**